index.html

```diff
@@ -49,19 +49,19 @@ <h1 class="large l-align-center l-mb-mega">&lt;gaia-button&gt;</h1>
   <hr/>
 
   <section>
-    <gaia-button><i data-icon="back-light" aria-hidden="true"></i><span>Back</span></gaia-button>
-    <gaia-button><span>Forward</span><i data-icon="forward-light" aria-hidden="true"></i></gaia-button>
+    <gaia-button><span data-l10n-id="back" data-icon="back-light">Back</span></gaia-button>
+    <gaia-button><span data-l10n-id="forward" data-icon-end="forward-light">Forward</span></gaia-button>
   </section>
 
   <hr/>
 
   <section flex>
-    <gaia-button circular><i data-icon="camera"></i></gaia-button>
-    <gaia-button circular><i data-icon="bug"></i></gaia-button>
-    <gaia-button circular><i data-icon="video"></i></gaia-button>
-    <gaia-button circular><i data-icon="camera"></i></gaia-button>
-    <gaia-button circular><i data-icon="reload"></i></gaia-button>
-    <gaia-button circular><i data-icon="airplane"></i></gaia-button>
+    <gaia-button circular><span data-l10n-id="camera" data-icon="camera"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="bug" data-icon="bug"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="video" data-icon="video"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="camera" data-icon="camera"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="reload" data-icon="reload"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="airplane" data-icon="airplane"/></gaia-button>
```

Comment:
Yeah it looks good. As I mentioned in https://github.com/gaia-components/gaia-button/pull/12/files#r31928864, in this case data-l10n-id value will likely point to 2 entries (if button has text, otherwise 1) in properties file for example:
```
<gaia-button data-icon="dicon" data-l10n-id="l10nId"></button>
// will point to
l10nId = Button text if needed
l10nId.ariaLabel = One of: icon label (if no text), icon label + button text (if both and icon is relevant), button text (if icon is irrelevant)
```